cmd/ztest/ztest.c

```diff
@@ -6845,6 +6888,11 @@ ztest_run(ztest_shared_t *zs)
 	if (ztest_opts.zo_verbose >= 4)
 		(void) printf("starting main threads...\n");
 
+
+	/* Replay all logs of all datasets in the pool */
```

Comment:
Since `ztest_dataset_open()` will also do this replay, let's make it clear this is primarily for the temporary datasets which otherwise won't get replayed, which can in turn results in the reported failure.